app/app.go

```diff
@@ -126,14 +128,66 @@ func (a *App) Run(ctx context.Context) (retErr error) {
 	// Leader election
 	if a.LeaderElectionOptions.LeaderElect {
 		a.Logger.Info("Starting leader election", logz.NamespaceName(a.LeaderElectionOptions.ConfigMapNamespace))
-		ctx, err = DoLeaderElection(ctx, a.Logger, a.Name, a.LeaderElectionOptions, a.MainClient.CoreV1(), recorder)
```

Comment:
@amckague you haven't deleted that method - the code will not compile with old libraries. We need tests...